daemon/api_prompting_test.go

```diff
@@ -252,6 +255,392 @@ func (s *promptingSuite) TestGetUserID(c *C) {
+					"message": prompting_errors.ErrPromptNotFound.Error(),
+					"kind":    string(client.ErrorKindInterfacesRequestsPromptNotFound),
+				},
+				"status":      "Not Found",
+				"status-code": 404.0,
+				"type":        "error",
+			},
+		},
+		{
+			err: prompting_errors.ErrRuleNotFound,
+			body: map[string]interface{}{
+				"result": map[string]interface{}{
+					"message": prompting_errors.ErrRuleNotFound.Error(),
+					"kind":    string(client.ErrorKindInterfacesRequestsRuleNotFound),
+				},
+				"status":      "Not Found",
+				"status-code": 404.0,
+				"type":        "error",
+			},
+		},
+		{
+			err: prompting_errors.ErrRuleNotAllowed,
+			body: map[string]interface{}{
+				"result": map[string]interface{}{
+					"message": prompting_errors.ErrRuleNotAllowed.Error(),
+					"kind":    string(client.ErrorKindInterfacesRequestsRuleNotFound),
+				},
+				"status":      "Not Found",
+				"status-code": 404.0,
+				"type":        "error",
+			},
+		},
+		{
+			err: prompting_errors.ErrPromptsClosed,
+			body: map[string]interface{}{
+				"result": map[string]interface{}{
+					"message": prompting_errors.ErrPromptsClosed.Error(),
+				},
+				"status":      "Internal Server Error",
+				"status-code": 500.0,
+				"type":        "error",
+			},
+		},
+		{
+			err: prompting_errors.ErrRulesClosed,
+			body: map[string]interface{}{
+				"result": map[string]interface{}{
+					"message": prompting_errors.ErrRulesClosed.Error(),
+				},
+				"status":      "Internal Server Error",
+				"status-code": 500.0,
+				"type":        "error",
+			},
+		},
+		{
+			err: prompting_errors.ErrTooManyPrompts,
+			body: map[string]interface{}{
+				"result": map[string]interface{}{
+					"message": prompting_errors.ErrTooManyPrompts.Error(),
+				},
+				"status":      "Internal Server Error",
+				"status-code": 500.0,
+				"type":        "error",
+			},
+		},
+		{
+			err: prompting_errors.ErrRuleIDConflict,
+			body: map[string]interface{}{
+				"result": map[string]interface{}{
+					"message": prompting_errors.ErrRuleIDConflict.Error(),
+				},
+				"status":      "Internal Server Error",
+				"status-code": 500.0,
+				"type":        "error",
+			},
+		},
+		{
+			err: prompting_errors.ErrRuleDBInconsistent,
+			body: map[string]interface{}{
+				"result": map[string]interface{}{
+					"message": prompting_errors.ErrRuleDBInconsistent.Error(),
+				},
+				"status":      "Internal Server Error",
+				"status-code": 500.0,
+				"type":        "error",
+			},
+		},
+		{
+			err: prompting_errors.NewInvalidOutcomeError("foo", []string{"bar", "baz"}),
+			body: map[string]interface{}{
+				"result": map[string]interface{}{
+					"message": `invalid outcome: "foo"`,
+					"kind":    "interfaces-requests-invalid-fields",
+					"value": map[string]interface{}{
+						"outcome": map[string]interface{}{
+							"reason":    "unsupported-value",
+							"supported": []interface{}{"bar", "baz"},
+							"value":     "foo",
```

Comment:
As we chatted about on our call earlier @olivercalder, I think so long as we have this `value` field as an array of strings in all cases I think we're good from a structural point of view with the API responses :slightly_smiling_face: